cpp/CMakeLists.txt

```diff
@@ -851,6 +864,9 @@ add_dependencies(arrow_test_dependencies toolchain-tests)
 if(ARROW_STATIC_LINK_LIBS)
   add_dependencies(arrow_dependencies ${ARROW_STATIC_LINK_LIBS})
   if(ARROW_HDFS OR ARROW_ORC)
+    if(Protobuf_SOURCE STREQUAL "SYSTEM")
```

Comment:
@kou This seems to require protobuf when HDFS is enabled, is there any reason for it? Is it a mistake?

Reply:
Oh, good catch! It might be a mistake. Let's try removing this: #39136